core/thread/src/TThreadExecutor.cxx

```diff
@@ -1,15 +1,32 @@
 #include "ROOT/TThreadExecutor.hxx"
 #include "tbb/tbb.h"
+#include "TROOT.h"
+#include <iostream>
 
 namespace ROOT{
+
+  unsigned TThreadExecutor::fPoolSize = 0;
+
   TThreadExecutor::TThreadExecutor():fInitTBB(new tbb::task_scheduler_init()){
   }
 
   TThreadExecutor::TThreadExecutor(size_t nThreads):fInitTBB(new tbb::task_scheduler_init(nThreads)){
+    if (fPoolSize != 0){
+      auto cause = "TThreadExecutor";
+      if(ROOT::IsImplicitMTEnabled()){
+        fPoolSize = ROOT::GetImplicitMTPoolSize();
+        fPoolSize = 3;
+        cause = "ROOT::EnableImplicitMT";
+      }
+      std::cout<<"Can't change the number of threads specified by a previous "<<cause<<" instantiation. Proceding with "<<fPoolSize<<" threads"<<std::endl;
```

Comment:
The experiments like our error / warning diagnostic messages to be capturable. That's why we generally use Error() from TError.h. Could you do that, too, throughout your code?